### ISSUE TITLE
Tests on Xiaofang S1

### ISSUE THREAD

**Issue by a2ng** [EliasKotlyar/Xiaomi-Dafang-Hacks#349]
Here's what i just done with my brand new *Xiaofang** S1* ::
1. Flash cfw with hack,  xiaofang/cfw-1.0.bin and after that successfully execute sdcard run.sh
2. modify run.sh to correct load wlan driver ::
 `insmod /driver/8189fs.ko rtw_initmac="$MAC"`
 and successfully connected to WiFi
here's the startup.log content ::
`
==================================================
Starting the Dafang Hacks Custom Application Layer
==================================================
loaded common functions
Bind mounted /system/sdcard/root to /root
Bind mounted /system/sdcard/etc to /etc
wpa_supplicant: Successfully initialized wpa_supplicant
rfkill: Cannot open RFKILL control device
nl80211: Could not re-add multicast membership for vendor events: -2 (No such file or directory)
udhcpc: udhcpc (v1.22.1) started
Sending discover...
Sending discover...
Sending select for 192.168.0.80...
Lease of 192.168.0.80 obtained, lease time 600
deleting routers
adding dns 192.168.0.1
adding dns 8.8.8.8
initialized gpios
dropbear:
bftpd:
lighttpd:
Startup finished!`

Now the problem is on "v4l2rtspserver-master", it won't start, i try to run manually and here's the result ::
`failed to AddSensor
IMP_System_Init() failed
Segmentation fault`

so, it looks like need help here.

Thanks.

**Comment by jmtatsch** [comment]
Possibly try the binary from commit https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/commit/62712d932c6326f9dccea0e4c7ac19786304852e it seems to have solved other peoples IMP_System_Init() failed problem.

**Comment by a2ng** [comment]
still not working.

**Comment by eigger** [comment]
I am also Xiaofang S1. But it does not work.
I tried 
1) binary from commit 62712d9. 
2) cfw1.0 / cfw1.1 / cfw1.2

**Comment by jmtatsch** [comment]
@eigger not sure you are using the right custom firmware. As far as i can see there is only cfw-1.0.bin https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/blob/master/hacks/cfw/xiaofang/cfw-1.0.bin for the xiaofang.

**Comment by jmtatsch** [comment]
Also see this related issue #223 for more hints towards solving the IMP_System_Init() failed Segmentation fault.

**Comment by eigger** [comment]
@jmtatsch  I tried Wyze because Xiaofang did not work.
Maybe it is my mistake.
I will try again. Thank you.

**Comment by eigger** [comment]
@jmtatsch 
I tried again. But does not work.
1) Xiaofang firmware don't have rtl8189ftv.ko driver(have 8189fs.ko, 1S Use rtl8189ftv.ko)
2) Wyze firmware have rtl8189ftv.ko but RTSP not work.(same IMP_System_Init() fail)
3) Wyze rtl8189ftv.ko driver copy to SD. 
    Change to Xiaofang frimware 
    Load from SD(rtl8189ftv.ko). but RTSP not work.(IMP_System_Init()  fail) 
I have read the #223 issue, but I do not work from the beginning.
Are Xiaofang and Xiaofang 1S different? Orz..

**Comment by a2ng** [comment]
@jmtatsch 
is there anything i do to help with IMP_System_Init() fail error on my Xiaofang S1 ?

**Comment by jmtatsch** [comment]
You can add some debug statements around the area in question and compile yourself.

**Comment by nik0** [comment]
What you can try if you recompile by yourself:
remove the libimp.a dependency when compiling v4l2master and use the one you will find on the camera.
It doesn't work on the Dafang but I never try to find out why ...

**Comment by youqingkui** [comment]
me too

==================================================
Starting the Dafang Hacks Custom Application Layer
==================================================
loaded common functions
Created etc directory on sdcard
Bind mounted /system/sdcard/root to /root
Bind mounted /system/sdcard/etc to /etc
Created cron directory
wpa_supplicant: Successfully initialized wpa_supplicant
rfkill: Cannot open RFKILL control device
Could not read interface wlan0 flags: No such device
nl80211: Driver does not support authentication/association or connect commands
nl80211: deinit ifname=wlan0 disabled_11b_rates=0
Could not read interface wlan0 flags: No such device
wlan0: Failed to initialize driver interface
udhcpc: 
initialized gpios
dropbear: 
bftpd: 
created new certificate for webserver
lighttpd: 
Startup finished!

**Comment by EliasKotlyar** [comment]
Hello everyone,

I have an idea, to resolve this problemn. I suppose that i know what happens: The Xiaofang 1S is using a different sensor than a normal Dafang. To confirm this theory I need some help:
Can someone who have a S1 and serial-headers attached run "lsmod" on original firmware(not CFW)? I suppose that it has a different sensor driver running.

**Comment by a2ng** [comment]
to @EliasKotlyar 
i don't know how to attach serial-headers? but i have logged with CFW and here's lsmod, content of /drivers  and dmesg result.
[lsmod.txt](https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/files/2020308/lsmod.txt)
i hope this will helpfull

**Comment by EliasKotlyar** [comment]
PS: Also try this:
echo 1 >/proc/jz/sinfo/info; 
cat /proc/jz/sinfo/info

**Comment by a2ng** [comment]
to @EliasKotlyar 

i use latest commit and when execute sample_sinfo ::
`
[root@DAFANG:driver]# sample_sinfo
-sh: sample_sinfo: not found
`
and also
`
[root@DAFANG:bin]# ./sample_sinfo
-sh: ./sample_sinfo: not found
`
and here's latest startup.log ::
`
Starting the Dafang Hacks Custom Application Layer
==================================================
Loaded common functions
Bind mounted /system/sdcard/root to /root
Bind mounted /system/sdcard/etc to /etc
wpa_supplicant: Successfully initialized wpa_supplicant
rfkill: Cannot open RFKILL control device
nl80211: Could not re-add multicast membership for vendor events: -2 (No such fi
udhcpc: udhcpc (v1.22.1) started
Sending discover...
Sending discover...
Sending discover...
Sending select for 192.168.1.82...
Sending select for 192.168.1.82...
Lease of 192.168.1.82 obtained, lease time 600
deleting routers
adding dns 192.168.0.1
adding dns 8.8.8.8
Initialized gpios
dropbear:
bftpd:
lighttpd:
Startup finished!
`
i also modify to insmod /driver/8189fs.ko on run.sh on

**Comment by a2ng** [comment]
to @EliasKotlyar 
... and there's no /proc/jz/sinfo/info
here's dmsg text :: 
[sample_sinfo_dmsg.txt](https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/files/2028869/sample_sinfo_dmsg.txt)

**Comment by EliasKotlyar** [comment]
Okey. Can you please update to the latest version(only rootfs, not microsd-bootloader) and then try to run
```
/system/sdcard/bin/sample_sinfo
```
Also 
```
echo 1 >/proc/jz/sinfo/info;
cat /proc/jz/sinfo/info
```
Should work after this line of code:
```
insmod /system/sdcard/driver/sinfo
```

**Comment by a2ng** [comment]
To @EliasKotlyar 
i use the latest "firmware_mod" (is this correct for this ?? ...update to the latest version(only rootfs, not microsd-bootloader)...)  with some modification on run.sh to work ::
`
insmod /driver/8189fs.ko
`
and
`
insmod /system/sdcard/driver/sinfo.ko
`
because it doest'n run without '.ko'
and then run 
`
/system/sdcard/bin/sample_sinfo
`
result is still same 
`
-sh:  /system/sdcard/bin/sample_sinfo: not found
`
and also run 
`echo 1 >/proc/jz/sinfo/info;
cat /proc/jz/sinfo/info
`
result is 
`
sensor not found
`

**Comment by EliasKotlyar** [comment]
Hello @a2ng ,

I have compiled a new Version of v4l2server. Can you please test? Try to mount the proper driver for your camera(i assume its a jxh23):
```
insmod /system/sdcard/driver/sensor_jxf23.ko data_interface=2 pwdn_gpio=-1 reset_gpio=18 sensor_gpio_func=0
```

**Comment by a2ng** [comment]
Thanks for your good work @EliasKotlyar 

i've done change on run.sh to sensor_jxf23.ko,
v4l2rtspserver-master still unsuccessfull to run, here's content of v4l2rtspserver-master.log
`/system/sdcard/bin/v4l2rtspserver-master: can't load library 'libmp3lame.so.0'
`

**Comment by a2ng** [comment]
@EliasKotlyar 
still not, here's content of v4l2rtspserver-master.log
`
1970-01-01 08:00:17.748 (   2.299s) [main thread     ]         ImpEncoder.cpp:1167     0| ##### sensor not found

1970-01-01 08:00:17.756 (   2.307s) [main thread     ]         ImpEncoder.cpp:1180     0| Found Sensor with ID:-1
1970-01-01 08:00:17.795 (   2.346s) [main thread     ]         ImpEncoder.cpp:1207   ERR| failed to AddSensor
1970-01-01 08:00:17.795 (   2.346s) [main thread     ]         ImpEncoder.cpp:830    ERR| IMP_System_Init() failed
`

**Comment by a2ng** [comment]
@EliasKotlyar 
i think there's no file `/system/sdcard/driver/sensor_jxf23.ko` ?

**Comment by EliasKotlyar** [comment]
ah yes. Please try with 
```
insmod /driver/sensor_jxf23.ko data_interface=2 pwdn_gpio=-1 reset_gpio=18 sensor_gpio_func=0
```

**Comment by a2ng** [comment]
@EliasKotlyar 
still not, v4l2rtspserver-master.log result
`
1970-01-01 08:00:18.767 (   2.301s) [main thread     ]         ImpEncoder.cpp:1167     0| ##### sensor not found

1970-01-01 08:00:18.771 (   2.304s) [main thread     ]         ImpEncoder.cpp:1180     0| Found Sensor with ID:-1
`

**Comment by EliasKotlyar** [comment]
This is not a error - it cannot determine the sensor name, and it then it uses jxf23. I suppose it should work, despite of this error. If not: Can you tell me what happens? Can you post the complete log?

**Comment by a2ng** [comment]
@EliasKotlyar 
i guest that is, but Sorry, i have no idea, here's dmesg
[latest_dmesg.txt](https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/files/2051582/latest_dmesg.txt)

**Comment by a2ng** [comment]
to @EliasKotlyar 
I guest you have to recompile v4l2rtspserver-master with condition xiaofang s1 with sensor name `jxf23`  and dafang `jxf22`

thanks.

**Comment by APellissier** [comment]
Hello, I have exactly the same problem. Have you find a solution?
Thanks you

**Comment by drcobra** [comment]
Hi,
I am having the same problem. This is my log in case it is any help:
[dsmeg.txt](https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/files/2059684/dsmeg.txt)

Thank you

**Comment by APellissier** [comment]
If i resume that i understand in the post.
I've copy the missing driver for jxf23 (i downloaded this file https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/blob/master/firmware_original/xiaofang/5.6.2.43/driver.bin and i haved extract it with 7zip)
in driver folders on sdcard.
Now the error "failed sensor find" is disapear.
But an error occur with v4l2rtspserver-master.
@a2ng 
You say recompile v4l2rtspserver, you have try and solved the problem?

Thank you

**Comment by a2ng** [comment]
@APellissier 
so far i get that xiaofang s1, use sensor_jxf23 and you i use directly from /driver/ same as 8189fs.
v4l2rtspserver found sensor driver but still load jxf22, now i just prepare to compile from new repository, and i think i need guides from the experts.
To @nik0 maybe you can share how compile, is just clone git repository to xiofang and run compile.sh ? or can be done somewhere else (i try on my ARM raspberry pi and stuck since codea ar for MIPS).
Thanka.

**Comment by nik0** [comment]
To compile you need a Linux machine (x86). I use a virtual machine on Windows.
Then clone the repo and run compile.sh

**Comment by a2ng** [comment]
ok thanks @nik0 i'try

**Comment by a2ng** [comment]
thanks to @nik0 
i've succeded compile Main repository on my VM using ubuntu,
but still have no idea to work with sensor_jxf23 on my xiaofang 1s.
is that possible to disable sensor by ::
- remove insmod sensor_jxf23.ko
- on ImpEncoder.cpp, disable IMP_ISP_AddSensor, IMP_ISP_Enable

**Comment by nik0** [comment]
Are you sure it is an jxf23 ?
Per your logs the sensor is not found
`ImpEncoder.cpp:1188     0| Found Sensor with ID:-1` so first you need to figure out what is sensor ...

**Comment by a2ng** [comment]
to @nik0 and @EliasKotlyar 
i have made progress, overwriting tx-isp.ko file from /driver to /system/sdcard/driver, now video and audio working fine. on dmesg sinfo still `info: failed sensor find` on jxf23, and v4l2rtspserver-master.log throws a lot of message. At first line shows `ERR| IMP_ISP_EnableTuning failed` so will raise `ERR| IMP_System_Init() failed`.  i've attach those log files,
Thanks.
[v4l2rtspserver-master.log](https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/files/2065721/v4l2rtspserver-master.log)
[xiaofang_1s_dmeg_working.txt](https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/files/2065722/xiaofang_1s_dmeg_working.txt)

**Comment by nik0** [comment]
do you have more logs running "logcat" ?

**Comment by a2ng** [comment]
@nik0 
here's 
[logcat_working.txt](https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/files/2065754/logcat_working.txt)

and when i disable motion detection from web interface, reboot, here's 
[v4l2rtspserver-master_no_motion.log](https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/files/2065755/v4l2rtspserver-master_no_motion.log)

**Comment by nik0** [comment]
@a2ng,
Can you summarize what is not working ?
I see you don't have the last version of v4l2rtp. This is not a big deal, but can you give me the result of "env" command.
You need to use the libimp.so that is in /system/sdcard/libs

**Comment by a2ng** [comment]
@nik0 
i have not full analize, just conclusion that jxf23 needs first run compatible tx-isp.ko, from here would you give me info, on dafang what's the different between /driver/tx-isp.ko with /system/sdcard/driver/tx-isp.ko ?
for v4l2rtp, all i know it base on https://github.com/mpromonet/v4l2rtspserver, would you share your latest ?
i try insmode /system/sdcard/lib/libimp.so but result is `...invalid module format`
here's my env 
`[root@DAFANG:~]# env
SSH_CLIENT=192.168.1.80 4189 22
USER=root
LD_LIBRARY_PATH=/thirdlib:/system/lib
HOME=/root
SSH_TTY=/dev/pts/0
PS1=[\u@\h:\W]#
LOGNAME=root
TERM=xterm
PATH=/system/sdcard/bin:/system/bin:/bin:/sbin:/usr/bin:/usr/sbin
SHELL=/bin/sh
PWD=/root
SSH_CONNECTION=192.168.1.80 4189 192.168.1.81 22
`
Thanks.

**Comment by nik0** [comment]
The sources that we use are here https://github.com/Dafang-Hacks/Main/tree/master/v4l2rtspserver-master
You don't need to "insmod", this is not a driver.
before starting the server try this command:
`export LD_LIBRARY_PATH=/system/sdcard/lib:/thirdlib:/system/lib `

**Comment by APellissier** [comment]
i have a serial hearer connected to my xiaofang but i'm newbie on linux.
that you ask is to connect in serial with the original firmware and extract the files localized  in / dev?
thanks for your help

**Comment by EliasKotlyar** [comment]
Please do the following steps:
1. Boot into original Software, connect via Serial to get a Terminal Connection(Refer here:https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/blob/master/hacks/serial.md)
2. Run "lsmod" and paste the output here
3. Go into /sys/module/ and make "ls". Paste the output here
5. Go into /sys/module/<sensorname> /parameters/ and "cat" every file there. Paste the output here. Sensorname may vary. You should find it easily. It should be sensor_jxf23 or sensor_jxf22.

Sample output:
```
[root@DAFANG:parameters]# pwd
/sys/module/sensor_jxf22/parameters
[root@DAFANG:parameters]# ls -lha
total 0
drwxr-xr-x    2 root     root           0 Jun  4 03:35 .
drwxr-xr-x    7 root     root           0 Jun  4 03:30 ..
-r--r--r--    1 root     root        4.0K Jun  4 03:35 data_interface
-r--r--r--    1 root     root        4.0K Jun  4 03:35 pwdn_gpio
-r--r--r--    1 root     root        4.0K Jun  4 03:35 reset_gpio
-r--r--r--    1 root     root        4.0K Jun  4 03:35 sensor_gpio_func
[root@DAFANG:parameters]# cat pwdn_gpio 
-1
[root@DAFANG:parameters]# cat reset_gpio 
18
[root@DAFANG:parameters]# cat sensor_gpio_func 
0
[root@DAFANG:parameters]# cat data_interface 
2
```

**Comment by APellissier** [comment]
This is the result:

`[root@Ingenic-uc1_1:parameters]# lsmod
sensor_jxf23 9120 1 - Live 0xc05a7000 (O)
rtl8189ftv 957549 0 - Live 0xc0425000 (O)
sample_pwm_hal 3592 2 - Live 0xc02de000 (O)
sample_pwm_core 2702 1 - Live 0xc02d7000 (O)
sinfo 9600 0 - Live 0xc02ce000 (O)
audio 60341 2 - Live 0xc02b4000 (O)
sample_motor 11647 0 - Live 0xc0296000 (O)
exfat 106720 0 - Live 0xc026f000 (O)
tx_isp 363556 4 - Live 0xc01e4000 (O)


[root@Ingenic-uc1_1:parameters]# ls /sys/module/
audio            jzmmc_v12        rtl8189ftv       tcp_cubic
block            kernel           sample_motor     tx_isp
brd              lockd            sample_pwm_core  usb_storage
cfg80211         loop             sample_pwm_hal   usbcore
configfs         mac80211         scsi_mod         usbhid
dwc2             mmc_core         sensor_jxf23     videobuf2_core
exfat            mmcblk           sg               workqueue
firmware_class   nfs              sinfo            xz_dec
hid              printk           soundcore        zd1201
i2c_algo_bit     random           spurious
i8042            rcupdate         sunrpc
jz_nvpu          rcutree          sysrq


[root@Ingenic-uc1_1:parameters]# pwd
/sys/module/sensor_jxf23/parameters

[root@Ingenic-uc1_1:parameters]# cat pwdn_gpio
-1
[root@Ingenic-uc1_1:parameters]# cat reset_gpio
18
[root@Ingenic-uc1_1:parameters]# cat sensor_gpio_func
0
[root@Ingenic-uc1_1:parameters]# cat data_interface
2`

**Comment by EliasKotlyar** [comment]
Okey , it seems that we are out of luck here. The driver is mapped correctly with the right parameters. 
I dont have a lot of ideas left, but we can try to make everything look like the original firmware

Can you modify your run.sh as follows:

```
## Start Sensor:
insmod /driver/tx-isp.ko isp_clk=100000000
insmod /driver/sensor_jxf23.ko data_interface=2 pwdn_gpio=-1 reset_gpio=18 sensor_gpio_func=0
insmod /driver/sinfo.ko
```

The difference is the tx_isp driver, which was loaded from the sdcard. maybe it will solve this error by using the original one.

**Comment by a2ng** [comment]
i agree. so for now we must use those 3 driver files to work.
so far my xiaodang 1s all runs as dafang, can hear audio also.
for @EliasKotlyar  
i have question, on dafang why we use /system/sdcard/driver ?
I saw jxf22, tx-isp was different on /driver with /system/sdcard/driver who came from cfw.
Thanks.

**Comment by stve-a** [comment]
Hi all, 
First of all wanna say I really appreciate that some of you guys are trying to make this work for the xiaofang S1.
I'm not so much a technical guy, so a lot that is said in this thread is chinese to me (pun intended ;) )

I was just wondering if anyone already got the RTSP stream part working on the FANG S1, and if so ... how.

Keep up the good work guys, know that it is appreciated !!

Regards
S

**Comment by LouiS22** [comment]
@EliasKotlyar thanks for the patch, seems working with my 1S (camera now has image). However I have another issue: I'm not able to change any setting, and also no log files at all (could be a read only issue? I've tried to play with remount, no luck so far).

**Comment by stve-a** [comment]
@LouiS22 could you share the steps you did to make the video work ? (I assume the rstp stream works now) ?

regards
S

**Comment by LouiS22** [comment]
@stve-a the solution is above, Elias posted 4 hours ago. You have to replace 3 lines in the run.sh, that should be all. I've tested, RTSP working.

**Comment by stve-a** [comment]
@LouiS22 so no special build or anything, just the master and the 3 line replace.
Will try it now, thanks

**Comment by LouiS22** [comment]
@stve-a make a backup though, as you can see I have a read only issue right now (probably unrelated to the current master, but who knows?)

**Comment by stve-a** [comment]
I got RTSP working now also with the 3 line replace in run.sh

Keep up the good work !
Awesome.

**Comment by LouiS22** [comment]
@stve-a everything else works? Do you have visible logs, can change settings, etc?

**Comment by stve-a** [comment]
@LouiS22  well the website looks weird. I can logon but cannot change anything.
I have created an issue for this an hour ago or so: https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/issues/410

I have this on both DAFANG and XIAOFANG S1
They both have exactly same issue with website.

Regards
S

**Comment by stve-a** [comment]
@LouiS22 Niko helped me get it right with the website thing. It was an error on my part.
So I tested making changes on the S1 and it works fine, I can change timezones and hostnames and so on.

Regards
S

**Comment by stve-a** [comment]
When I choose a resolution higher the 720p the RSTP stream stops, so it does not work in 1080p for me at least.  Anyone else having this ?

**Comment by APellissier** [comment]
Thank you for you help, now the video work :) but I have the same problem as LouiS22 :  I'm not able to change any setting, and also no log files at all.

**Comment by kar-vi-nen** [comment]
Hello,
Good work!

Although i have some problem with my 1S. I have serial access so i have attached [boot.log](https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/files/2069770/boot.log) captured with putty.
I'm afraid this 3 line fix isn't working for my device. :(

This device has "2018-03-02" printed in pcb so it's quite new. Is there a way to make sure that this device has jfx23, other than disassemble whole camera?

Whats that "Invalid Argument ?" line after Starting v4l2rtspserver-master ?
This is what worries me : [   28.415959] jxf23 0-0040: chip found @ 0x40 (i2c0) is not an jxf23 chip.

**Comment by EliasKotlyar** [comment]
@kar-vi-nen : Maybe you have a different sensor installed? Can you try with jxf22? Maybe its also something completely different. Can you make a "ls /driver/" and post the results here?

**Comment by EliasKotlyar** [comment]
Ok , I have added support for the 1S here. I close the issue for now, cause its getting really confusing here. We can continue the development discussion here:
https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/issues/411

**Comment by SamHead** [comment]
@kar-vi-nen

Did they reduce Dram?

My Dafang
U-Boot 2013.07 (Dec 21 2017 - 18:44:33)
Board: ISVP (Ingenic XBurst T20 SoC)
DRAM:  128 MiB

1S
U-Boot 2013.07 (Mar 13 2018 - 15:07:07)

Board: ISVP (Ingenic XBurst T20 SoC)
DRAM:  64 MiB

**Comment by tookus123** [comment]
Possible good news for Xiaofang 1s users that haven't been able to get rtsp working. Some Xiaofang 1s users don't have problems but for others like me, this is what i did. 

Replace the ##Start Sensor section in run.sh with what i have below. @EliasKotlyar - The modules loaded are the same as the Dafang Classic/Wyzecam V2. Looks like there might be two versions of the Xiaofang 1s? One that uses sensor_jxf22 and one that uses sensor_jxf23? Thanks for the great work on this project. 

## Start Sensor:
insmod /driver/tx-isp.ko isp_clk=100000000
insmod /driver/sensor_jxf22.ko data_interface=2 pwdn_gpio=-1 reset_gpio=18 sensor_gpio_func=0
insmod /system/sdcard/driver/sinfo.ko

**Comment by kar-vi-nen** [comment]
Oh, i didnt know it is possible to comment closed issue.
I finally got my camera working. There was loose cable between pcbs... 

Mine has 64M memory. Im not getting 1080p picture because of this. Im guessing there isn't simply enough memory for 1080p h246 even if i could change memory mapping from uboot.

**Comment by Changtaek** [comment]
My Xiaofang don't start v4l2rtspserver-master
My logs
[dmesg_etc.txt](https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/files/2182848/dmesg_etc.txt)

**Comment by freber** [comment]
Can anyone share the /driver/sensor_jxf23.ko file? I'm not able to find a way to extract it from drivers.bin (I'm on a Mac and don't have 7zip).

**Comment by kiddyfurby** [comment]
My 1S has a 2018-03-02 PCB

```
[root@DAFANG:~]# ls /driver/
audio.ko              rtl8189ftv.ko         sample_pwm_core.ko    sample_speakerctl.ko  sinfo.ko
exfat.ko              sample_motor.ko       sample_pwm_hal.ko     sensor_jxf22.ko       tx-isp.ko
```
```
[root@DAFANG:~]# cat /proc/meminfo 
MemTotal:          35300 kB
```
Tons of i2c errors in my dmesg
```
[   14.656191] name : i2c-gpio1 nr : 1
[   14.659791] name : i2c0 nr : 0
[   14.726135] i2c i2c-0: i2c_jz_irq 441, I2C transfer error, ABORT interrupt
[   14.733250] i2c i2c-0: --I2C txabrt:
[   14.736937] i2c i2c-0: --I2C TXABRT[0]=I2C_TXABRT_ABRT_7B_ADDR_NOACK
[   14.743511] error: sensor_read,172 ret = -5
[   14.747818]  sensor_read: addr=0xa value = 0x0
[   14.752407] err sensor read addr = 0xa, value = 0x0
[   14.817617] i2c i2c-0: i2c_jz_irq 441, I2C transfer error, ABORT interrupt
[   14.824724] i2c i2c-0: --I2C txabrt:
[   14.828413] i2c i2c-0: --I2C TXABRT[0]=I2C_TXABRT_ABRT_7B_ADDR_NOACK
[   14.834985] error: sensor_read,172 ret = -5
[   14.839360]  sensor_read: addr=0x300a value = 0x0
[   14.844551] err sensor read addr = 0x300a, value = 0x0
[   14.909882] i2c i2c-0: i2c_jz_irq 441, I2C transfer error, ABORT interrupt
[   14.917066] i2c i2c-0: --I2C txabrt:
[   14.920861] i2c i2c-0: --I2C TXABRT[0]=I2C_TXABRT_ABRT_7B_ADDR_NOACK
[   14.927446] error: sensor_read,172 ret = -5
[   14.931771]  sensor_read: addr=0x300b value = 0x0
[   14.936615] err sensor read addr = 0x300b, value = 0x0
[   15.001923] i2c i2c-0: i2c_jz_irq 441, I2C transfer error, ABORT interrupt
```
```
[root@DAFANG:~]# echo 1 >/proc/jz/sinfo/info 
[root@DAFANG:~]# cat /proc/jz/sinfo/info 
sensor not found
```
